tests/test_types.py

```
@@ -1883,15 +1883,22 @@ class Foo(BaseModel):
-        a: Fraction
+@pytest.mark.parametrize(
+    ['input', 'error_type'],
+    [
+        ('wrong_format', 'fraction_parsing'),  # Raises `ValueError` internally
+        ('1/0', 'fraction_parsing'),  # Raises `ZeroDivisionError` internally
+        (float('inf'), 'fraction_parsing'),  # Raises `OverflowError` internally
+        (float('nan'), 'fraction_parsing'),  # Raises `ValueError` internally
+        (type, 'fraction_type'),  # Raises `TypeError` internally
+    ],
+)
+def test_fraction_validation_error(input: Any, error_type: str):
```

Ruff (0.15.18)

[error] Function argument `input` is shadowing a Python builtin

(A002)